src/content/types.js

```diff
@@ -1,7 +1,25 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 // @flow
 
-import type { Store as ReduxStore } from 'redux'; // eslint-disable-line import/named
-import type { Action } from './actions/types';
-import type { State } from './reducers/types';
+/* eslint-disable import/named */
+import type {
+  Store as ReduxStore,
+  ThunkAction as ReduxThunkAction,
+  Dispatch as ReduxDispatch,
+  GetState as ReduxGetState,
+} from 'redux'; // eslint-disable-line import/named
+/* eslint-enable import/named */
+import type { Action as ActionsRef } from './actions/types';
+import type { State as StateRef } from './reducers/types';
 
+// Re-export these here so they are easily available from wherever and avoids
+// circular dependencies.
+export type Action = ActionsRef;
+export type State = StateRef;
+
+export type GetState = ReduxGetState<State>;
+export type ThunkAction<Result> = ReduxThunkAction<State, Action, Result>;
```

Comment:
It might be nice to do it this way:
```js
type AsyncThunk<T> = ThunkAction<Promise<T>>;
type SyncThunk<T> = ThunkAction<T>;
```
I think it would be valid to have thunks that don't involve async and promise behavior.